### PR TITLE
refactor: remove dead code in python client

### PR DESCRIPTION
**Pull request by Mause**
It's not possible to pass parameters to `__enter__` unless you call it directly, eg:

```python
conn.__enter__(read_only=True)
```

so it's not really useful having these parameters (and they just confuse type checkers)

### CODE REVIEW

File: tools/pythonpkg/src/pyconnection.cpp [hunk 1]
```diff
@@ -97,8 +97,7 @@ void DuckDBPyConnection::Initialize(py::handle &m) {
 	         "Serialize a query to protobuf on the JSON format", py::arg("query"))
 	    .def("get_table_names", &DuckDBPyConnection::GetTableNames, "Extract the required table names from a query",
 	         py::arg("query"))
-	    .def("__enter__", &DuckDBPyConnection::Enter, py::arg("database") = ":memory:", py::arg("read_only") = false,
-	         py::arg("config") = py::dict())
+	    .def("__enter__", &DuckDBPyConnection::Enter)
```
Comment (pdet) [inline tools/pythonpkg/src/pyconnection.cpp:100]: Can't this call connect directly then?

Comment (Mause) [reply tools/pythonpkg/src/pyconnection.cpp:100]: This is a method on the Connection class, so we already have a connection?

Comment (pdet) [reply tools/pythonpkg/src/pyconnection.cpp:100]: Ahh, is because ```__enter__``` requires a ```DuckDBPyConnection &self```. Alright, looks good.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**